sync-core/src/test/java/com/cloudant/sync/query/IndexTest.java

```diff
@@ -0,0 +1,135 @@
+        assertThat(index.indexType, is("text"));
+        assertThat(index.indexSettings.size(), is(1));
+        assertThat(index.indexSettings.get("tokenize"), is("simple"));
+    }
+
+    @Test
+    public void returnsNullWhenNoFields() {
+        Index index = Index.setUp(null, indexName);
+        assertThat(index, is(nullValue()));
+
+        index = Index.setUp(new ArrayList<Object>(), indexName);
+        assertThat(index, is(nullValue()));
+    }
+
+    @Test
+    public void returnsNullWhenNoIndexName() {
+        Index index = Index.setUp(fieldNames, null);
+        assertThat(index, is(nullValue()));
+
+        index = Index.setUp(fieldNames, "");
+        assertThat(index, is(nullValue()));
+    }
+
+    @Test
+    public void returnsNullWhenNoIndexType() {
+        Index index = Index.setUp(fieldNames, indexName, null);
+        assertThat(index, is(nullValue()));
+
+        index = Index.setUp(fieldNames, indexName, "");
+        assertThat(index, is(nullValue()));
+    }
+
+    @Test
+    public void returnsNullWhenInvalidIndexType() {
+        Index index = Index.setUp(fieldNames, indexName, "blah");
+        assertThat(index, is(nullValue()));
+    }
+
+    @Test
+    public void correctlyIgnoresIndexSettings() {
+        Map<String, String> indexSettings = new HashMap<String, String>();
+        indexSettings.put("tokenize", "porter");
+        // json indexes do not support index settings.  Index settings will be ignored.
+        Index index = Index.setUp(fieldNames, indexName, "json", indexSettings);
+        assertThat(index.indexSettings, is(nullValue()));
+    }
+
+    @Test
+    public void correctlySetsIndexSettings() {
+        Map<String, String> indexSettings = new HashMap<String, String>();
+        indexSettings.put("tokenize", "porter");
+        // text indexes support the tokenize setting.
+        Index index = Index.setUp(fieldNames, indexName, "text", indexSettings);
+        assertThat(index.indexSettings.size(), is(1));
+        assertThat(index.indexSettings.get("tokenize"), is("porter"));
+    }
+
+    @Test
+    public void comparesIndexTypeAndReturnsInEquality() {
+        Index index = Index.setUp(fieldNames, indexName);
+        assertThat(index.compareIndexTypeTo("text", null), is(false));
+    }
+
+    @Test
+    public void comparesIndexTypeAndReturnsEquality() {
+        Index index = Index.setUp(fieldNames, indexName);
+        assertThat(index.compareIndexTypeTo("json", null), is(true));
+    }
+
+    @Test
+    public void comparesIndexSettingsAndReturnsInEquality() {
+        Index index = Index.setUp(fieldNames, indexName, "text");
+        assertThat(index.compareIndexTypeTo("text", "{\"tokenize\":\"porter\"}"), is(false));
+    }
+
+    @Test
+    public void comparesIndexSettingsAndReturnsEquality() {
+        Index index = Index.setUp(fieldNames, indexName, "text");
+        assertThat(index.compareIndexTypeTo("text", "{\"tokenize\":\"simple\"}"), is(true));
+    }
```

Comment:
Related to a comment above, as `compareIndexTypeTo` takes a string, I would have comparison of semantically equivalent but different orderings:

```java
public void comparesIndexSettingsAndReturnsEquality() {
    Map indexSettings = /* Set up indexSettings Map, { "tokenise": "simple", "foo": "bar" } */;
    Index index = Index.setUp(fieldNames, indexName, "text", indexSettings);
    assertThat(index.compareIndexTypeTo("text", "{\"tokenize\":\"simple\", \"foo\":\"bar\"}"), is(true));
    assertThat(index.compareIndexTypeTo("text", "{\"foo\":\"bar\", \"tokenize\":\"simple\"}"), is(true));
}
```